strawberryfields/engine.py

```diff
@@ -74,11 +74,16 @@
 """
 
 import abc
+import uuid
 from collections.abc import Sequence
 from numpy import stack, shape
+from time import sleep
 
 from .backends import load_backend
-from .backends.base import (NotApplicableError, BaseBackend)
+from .backends.base import NotApplicableError, BaseBackend
```

Comment:
How come `BaseBackend` and `NotApplicableError` are being imported here? With the backends, we are currently trying to keep it as separate from the frontend as possible, with the aim of spinning them out into separate Python packages in the future (hence the use of the `load_backend` helper function).

Reply:
Looks like these are used in `LocalEngine` right now, but they lit up after running `black`. I'll look into this though.

Reply:
Ah, of course! Must have missed it in the previous PR.

Reply:
Seems like this may be fixed in the future by #111 , so will leave it as is for now.

Reply:
Looks like `BaseBackend` is only needed for isinstance checks. `_run_program` still catches `NotApplicableErrors` just in case, although the validation process during program compilation should prevent them.